articles/getting-started-using-admob-with-react-native-expo/index.md

```diff
@@ -0,0 +1,484 @@
+
+As we notice, we have rendered out the Banner ad at the top of the Home and Setting Screen. Now, we can move on to the Reward Ads.
+
+### STEP 4: Using Admob Reward Ads
+The Admob Reward Ad is a video ad that users need to watch to a certain length in order to obtain certain rewards. We can use it in multiple scenarios such as to receive game points, upgrading levels, etc.
+
+We are going to integrate this Reward Ad into the Setting screen. For this, we need to import the `AdMobRewarded` component from the expo-ads-admob package as shown in the code snippet below:
+
+```jsx
+import {
+  AdMobBanner,
+  AdMobRewarded,
+  setTestDeviceIDAsync,
+} from "expo-ads-admob";
+```
+
+Next, we need to create a new function to handle the Reward Ads. In the function, we need to set the Ad Unit ID, request, and show ads asynchronously. The overall configuration of the function is provided in the code snippet below:
+
+```jsx
+const initRewardAds = async () => {
+    // Display a rewarded ad
+    await AdMobRewarded.setAdUnitID("ca-app-pub-2547344479047582/6388494355");
+    await AdMobRewarded.requestAdAsync();
+    await AdMobRewarded.showAdAsync();
+  };
+```
+
+Now, we need to add a new Setting screen option to remove ads. It's going to be a `List` element item with a bullhorn icon. The template for it is provided in the code snippet below:
+
+```jsx
+<TouchableOpacity >
+        <List.Item
+          title="Remove Ads"
+          left={() => <List.Icon icon="bullhorn" />}
+        />
+ </TouchableOpacity>
+```
+
+We will get the Remove Ads list option as shown in the demo screenshot below:
+
+![add remove ad button](/engineering-education/getting-started-using-admob-with-react-native-expo/14-remove-ad-button.png)
+
+Next, we are going to create a `Modal` component. It will be used to ask the consent of the users for their agreement to watch video ads. For that, we need to import the required components as shown in the code snippet below:
+
+```jsx
+import {
+  View,
+  TouchableOpacity,
+  Text,
+  Modal,
+  StyleSheet,
+  Alert,
+} from "react-native";
+```
+
+First, we need to define a state using the `useState` hook in order to control the visibility of the `Modal` component as directed in the code snippet below:
+
+```jsx
+export default ({ navigation }) => {
+  const [modalVisible, setModalVisible] = useState(false);
+```
+
+Next, we need to add a `Modal` component with a body template. In the `Modal` body, we are going to add a button to trigger the Reward Ads as well. The Modal component with its prop configuration along with body template is provided in the code snippet below:
+
+```jsx
+<Modal
+        animationType="slide"
+        transparent={true}
+        visible={modalVisible}
+      >
+        <View style={styles.centeredView}>
+          <View style={styles.modalView}>
+            <Title style={styles.modalText}>
```

Comment:
The import statement for the Title component is missing from the code snippets. A person reading this article may not know that the component is from `react-native-paper`.

Reply:
updated as your mention.